app/lib/package/models.dart

```diff
@@ -353,6 +353,28 @@ class PackageVersionInfo extends db.ExpandoModel {
   }
 }
 
+/// Entity representing a package that has been removed.
+@db.Kind(name: 'ModeratedPackage', idType: db.IdType.String)
+class ModeratedPackage extends db.ExpandoModel {
+  @db.StringProperty(required: true)
+  String name;
+
+  @db.DateTimeProperty()
+  DateTime moderated;
+
+  /// The previous publisher id (null, if the package does not have a publisher).
+  @db.StringProperty()
+  String previousPublisherId;
```

Comment:
Not sure if we need to use the `previous` prefix on these names. It probably helps to distinguish it in a code search, but otherwise it could be just `publisherId` and `uploaders`, as by definition it stores the values that were set when the delete happened.

Reply:
I've had the same thought, so I'll remove the prefix.